### PR TITLE
Stories: A storytelling system to bring more life, drama, and soap opera plots into the station.

### PR DESCRIPTION
**Pull request by Iamgoofball**
dreamseeker_HGwaJjNrm5
dreamseeker_YhEiMngdFw
dreamseeker_8qWad8ewUC

:cl: Zonespace and Iamgoofball, passport sprites by Flavrius, stories written by Oscar Gilmour
add: Adds Stories to the game, a new system designed to enhance the roleplaying experience and breathe life, drama, and soap opera plots into the station!
/:cl:

# todo:

- [x] write more Unimpactful stories
- [x] write Somewhat Impactful stories
- [x] write Very Impactful stories
- [x] write Incredibly Impactful stories
- [x] get real sprites for passports
- [ ] code props for some of the story proposals
- [ ] test on the test server with ~~unpaid goons~~ willing playtesters
- [ ] have 30-50 stories ready to ship

### CODE REVIEW

**Comment by Paxilmaniac** [comment]
merge conflict any% speedrun

**Comment by Stalkeros2** [comment]
BASED
BAAAASEEEEEEEEED

**Comment by Kush1Push1** [comment]
Nice

**Comment by press-mark** [comment]
sick

File: _maps/map_files/generic/CentCom_skyrat_z2.dmm [hunk 1]
```diff
@@ -1045,9 +1045,9 @@
 	dir = 8;
 	dwidth = 1;
 	height = 13;
-	shuttle_id = "arrivals_shuttle";
 	name = "tram dock";
 	roundstart_template = /datum/map_template/shuttle/arrivals_skyrat;
+	shuttle_id = "arrivals_shuttle";
```
Comment (press-mark) [inline _maps/map_files/generic/CentCom_skyrat_z2.dmm:1050]: why was this bit touched

File: _maps/map_files/generic/CentCom_skyrat_z2.dmm [hunk 2]
```diff
@@ -10202,9 +10202,9 @@
 	dir = 8;
 	dwidth = 7;
 	height = 21;
-	shuttle_id = "goldeneye_cruiser_dock";
 	name = "Goldeneye Satellite Dock";
 	roundstart_template = /datum/map_template/shuttle/goldeneye_cruiser;
+	shuttle_id = "goldeneye_cruiser_dock";
```
Comment (press-mark) [inline _maps/map_files/generic/CentCom_skyrat_z2.dmm:10207]: same here i guess

**Comment by Stalkeros2** [comment]
Passport sprites from and made by our development team's spriter Flavrius.

File: modular_skyrat/modules/stories_system/code/helping_code/outfits.dm [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+	worn_id.update_label()
+	worn_id.update_icon()
+
+/datum/outfit/tourist
+	name = "Tourist"
+
+	id = /obj/item/card/id/passport
+	id_trim = /datum/id_trim/job/assistant/tourist
+	uniform = /obj/item/clothing/under/shorts/grey
+	suit = /obj/item/clothing/suit/hawaiian_blue
```
Comment (press-mark) [inline modular_skyrat/modules/stories_system/code/helping_code/outfits.dm]: these were changed just yesterday in https://github.com/Skyrat-SS13/Skyrat-tg/pull/17085/

File: modular_skyrat/modules/stories_system/code/helping_code/misc.dm [hunk 4]
```diff
@@ -0,0 +1,14 @@
+/obj/item/modular_computer/tablet/pda/centcom
+	name = "\improper Centcom PDA"
+	loaded_cartridge = /obj/item/computer_hardware/hard_drive/portable/command/captain
```
Comment (press-mark) [inline modular_skyrat/modules/stories_system/code/helping_code/misc.dm]:
```suggestion
	loaded_cartridge = /obj/item/computer_disk/command/captain
```

**Comment by Iamgoofball** [comment]
@Stalkeros2 updated the changelog thanks

File: _maps/map_files/generic/CentCom_skyrat_z2.dmm (hunk 2 above)
Comment (Zonespace27) [reply _maps/map_files/generic/CentCom_skyrat_z2.dmm:10207]: I touched CCz2 before switching over to the Z reservation system, this is just one of those things that updates when you touch its map file

**Comment by YakumoChen** [comment]
Tourists are goofs secret ploy to finally abolish communism and encourage capitalistic demands to undercut tourist wealth

**Comment by dawsonkeyes** [comment]
finally, americans have been added to ss13

**Comment by Iamgoofball** [comment]
I'll have some updates for this one on Monday, working double 12 hour shifts this weekend

File: modular_skyrat/modules/stories_system/code/helping_code/outfits.dm [hunk 5]
```diff
@@ -174,8 +174,37 @@
...
 		/obj/item/paper = 2,
 	)
+
+/datum/outfit/nri_shore_leave
+	name = "Shore Leave NRI Marine"
+
+	glasses = /obj/item/clothing/glasses/sunglasses
+	ears = /obj/item/radio/headset/guild
+	uniform = /obj/item/clothing/under/costume/nri
+	suit = /obj/item/clothing/suit/armor/vest/russian/nri
```
Comment (Stalkeros2) [inline modular_skyrat/modules/stories_system/code/helping_code/outfits.dm:299]: They don't really need the armor.

File: modular_skyrat/modules/stories_system/code/helping_code/outfits.dm [hunk 6]
```diff
@@ -174,8 +174,37 @@
...
+
+	back = /obj/item/storage/backpack/duffelbag/syndie/nri
+	backpack_contents = list(
+		/obj/item/storage/box/nri_survival_pack = 1,
+		/obj/item/crucifix = 1,
+		/obj/item/ammo_box/magazine/m9mm_aps = 1,
+		/obj/item/modular_computer/tablet/pda/security = 1,
+	)
+
+	l_pocket = /obj/item/gun/ballistic/automatic/pistol/ladon/nri
```
Comment (Stalkeros2) [inline modular_skyrat/modules/stories_system/code/helping_code/outfits.dm:312]: I'd prefer not to give them anything stronger than a Makarov; two-burst branded APS might be a little bit too much.

File: modular_skyrat/modules/stories_system/code/story_datums/somewhat_impactful.dm [hunk 7]
```diff
@@ -82,3 +82,20 @@
 	poor_sod = null
 	return ..()
 
+/*
+	NRI R&R
+		Plot Summary:
+			A small group of NRI marines are on shore leave, so they got drunk and decided to land on this station, ideally to get more drunk.
```
Comment (Stalkeros2) [inline modular_skyrat/modules/stories_system/code/story_datums/somewhat_impactful.dm:102]: I kinda wish they weren't military (I mean come on literally every NRI-station interaction is military-related good god this makes me very-very sad :( ) but this is very subjective and oh you do you.

**Review by Stalkeros2 — COMMENTED** [review]
Some NRI-related feedback by the unofficial official NRI lore dev.

File: modular_skyrat/modules/stories_system/code/story_datums/somewhat_impactful.dm (hunk 7 above)
Comment (Iamgoofball) [reply modular_skyrat/modules/stories_system/code/story_datums/somewhat_impactful.dm:102]: I can add a NRI tourist if we've got clothes for them

**Comment by Iamgoofball** [comment]
sorry for the delays, sick as fuck this week

File: modular_skyrat/modules/stories_system/code/story_datums/somewhat_impactful.dm (hunk 7 above)
Comment (Stalkeros2) [reply modular_skyrat/modules/stories_system/code/story_datums/somewhat_impactful.dm:102]: Like, we have an entire sprite pack of unused NRI diplomats' drip and an unfinished NRI diplomatic mission ERT. The latter could be removed and reused as a story instead. I think it'd fit way better than another NRI military group of pepol.

Comment (Iamgoofball) [reply modular_skyrat/modules/stories_system/code/story_datums/somewhat_impactful.dm:102]: sure, PR them to the branch and I'll accept it

Comment (Stalkeros2) [reply modular_skyrat/modules/stories_system/code/story_datums/somewhat_impactful.dm:102]: > sure, PR them to the branch and I'll accept it

Oh nice brb.
Relatively late response cuz GitHub went dead.

**Comment by Iamgoofball** [comment]
writers have begun work on this

File: modular_skyrat/modules/stories_system/code/helping_code/outfits.dm (hunk 5 above)
Comment (Lukaster42) [reply modular_skyrat/modules/stories_system/code/helping_code/outfits.dm:299]: > They don't really need the armor.

They do need the armor.

**Comment by Iamgoofball** [comment]
writers are working on this still, latest update was today, ETA of finishing the first batch should be soon :tm:

**Comment by Iamgoofball** [comment]
First batch of stories written by Oscar Gilmour are done, and are being added soon :tm:

**Pull request closed without merging**